src/solvers/exponentiate.jl

```diff
@@ -0,0 +1,48 @@
+function exponentiate_updater(
+  init;
+  psi_ref!,
+  PH_ref!,
+  outputlevel,
+  which_sweep,
+  region_updates,
+  which_region_update,
+  region_kwargs,
+  updater_kwargs,
+)
+  default_updater_kwargs = (;
+    krylovdim=30,  #from here only solver kwargs
+    maxiter=100,
+    outputlevel=0,
+    tol=1E-12,
+    ishermitian=true,
+    issymmetric=true,
+    eager=true,
+  )
+  updater_kwargs = merge(default_updater_kwargs, updater_kwargs)  #last collection has precedence
+  #H=copy(PH_ref![])
+  H = PH_ref![] ###since we are not changing H we don't need the copy
+  # let's test whether given region and sweep regions we can find out what the previous and next region were
+  # this will be needed in subspace expansion
+  #@show step_kwargs
+  substep = get(region_kwargs, :substep, nothing)
+  time_step = get(region_kwargs, :time_step, nothing)
+  @assert !isnothing(time_step) && !isnothing(substep)
+  region_ind = which_region_update
```

Comment:
```suggestion
  region_ind = which_region_update
```
I think we should avoid renaming variables like this, it makes it harder to read the code later on (i.e. fewer names are better).